docs/tutorial/flink.md

```diff
@@ -1,94 +1,407 @@
+   pod/flink-kubernetes-operator-559fccd895-pfdwj   2/2     Running   2 (2d21h ago)   6d17h
 
+   NAME                                     TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
+   service/flink-operator-webhook-service   ClusterIP   10.100.231.230   <none>        443/TCP             20d
+   service/kubernetes                       ClusterIP   10.100.0.1       <none>        443/TCP             20d
+   ```
 
-## Running the recipe
+## Setting up Runner Configuration
 
-Now run a recipe!
+Dataset recipes can be configured at runtime in a few different ways and with a couple
+of different configuration file formats
 
-```bash
-pangeo-forge-runner bake --repo <url-to-github-repo> --ref <name-of-branch-or-commit-hash>
-```
+### Runner Configuration
+
+Let's first look at defining *where* the data that our recipes will use is stored. There are three aspects of this question
+we must answer: 
+
+1. where data is discovered/read from
+
+2. where data is cached to, and 
+
+3. where data is output to
+
+### Where Data is Discovered
+
+Recipe input is defined in the recipe itself. Most recipes will define a `pangeo_forge_recipes.patterns.FilePattern` that provides the pipeline with input file locations.
+The example below taken from the [integration test recipe](https://github.com/pforgetest/gpcp-from-gcs-feedstock/blob/main/feedstock/recipe.py)
+
+   ```python
+   import apache_beam as beam
+   import pandas as pd
+
+   from pangeo_forge_recipes.patterns import ConcatDim, FilePattern
+   from pangeo_forge_recipes.transforms import OpenURLWithFSSpec, OpenWithXarray, StoreToZarr
+   
+   dates = [
+       d.to_pydatetime().strftime('%Y%m%d')
+       for d in pd.date_range("1996-10-01", "1999-02-01", freq="D")
+   ]
+
+   def make_url(time):
+       url_base = "https://storage.googleapis.com/pforge-test-data"
+       return f"{url_base}/gpcp/v01r03_daily_d{time}.nc"
+
+   concat_dim = ConcatDim("time", dates, nitems_per_file=1)
+   pattern = FilePattern(make_url, concat_dim)
+
+   recipe = (
+       beam.Create(pattern.items())
+       | OpenURLWithFSSpec()
+       | OpenWithXarray(file_type=pattern.file_type, xarray_open_kwargs={"decode_coords": "all"})
+       | StoreToZarr(
+           store_name="gpcp",
+           combine_dims=pattern.combine_dim_keys,
+       )
+   )
+   ```
+
+### Where Data is Cached
+
+The inputs and metadata can be cached so items don't have to be resolved again during the next recipe run. Defining where
+the caching happens is part of the runner file configuration. More about that in the [next session](#configfiles). Note 
+that in those upcoming examples keys `InputCacheStorage`` target these options.
+
+### Where Data is Output
+
+Defining where the data is output is also part of the runner file configuration. More about that in the [next session](#configfiles). Note 
+that in those upcoming examples the key `TargetStorage` defines this option.
+
+
+<a name="configfiles"></a>
```

Comment:
Since this uses myst markdown, yoou should use an [explicit target](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#creating-explicit-targets) here instead of a HTML `<a>` tag

Reply:
fixed